src/coreclr/vm/jitinterface.cpp

```diff
@@ -7142,6 +7142,27 @@ bool getILIntrinsicImplementationForInterlocked(MethodDesc * ftn,
     return true;
 }
 
+bool IsBitwiseEquatable(TypeHandle typeHandle, MethodTable * methodTable)
+{
+    if (!methodTable->IsValueType() || methodTable->ContainsPointers())
+    {
+        return false;
+    }
+
+    if (!CanCompareBitsOrUseFastGetHashCode(methodTable))
```

Comment:
Does this properly account for padding between struct members?

Reply:
Yes, it also handles floating point fields (rejects in case of them)